x/gamm/pool-models/balancer/pool.go

```diff
@@ -792,11 +796,6 @@ func (p *Pool) CalcJoinPoolNoSwapShares(ctx sdk.Context, tokensIn sdk.Coins, swa
 		return sdk.ZeroInt(), sdk.NewCoins(), err
 	}
 
-	// ensure that there aren't too many or too few assets in `tokensIn`
-	if tokensIn.Len() != p.NumAssets() {
-		return sdk.ZeroInt(), sdk.NewCoins(), errors.New("no-swap joins require LP'ing with all assets in pool")
-	}
-
```

Comment:
Note to reviewer: this was moved up so that both pool models return the same error given the same inputs